src/extension/executors/runner.ts

```diff
@@ -0,0 +1,163 @@
+import path from 'node:path'
+
+import { 
+  NotebookCellOutputItem, 
+  NotebookCellOutput, 
+  Task, 
+  TaskScope, 
+  CustomExecution, 
+  TaskRevealKind, 
+  TaskPanelKind, 
+  tasks, 
+  NotebookCellExecution,
+  TextDocument
+} from 'vscode'
+
+import { OutputType } from '../../constants'
+import { CellOutputPayload } from '../../types'
+import { PLATFORM_OS } from '../constants'
+import { IRunner, IRunnerEnvironment } from '../runner'
+import { getAnnotations, getCmdShellSeq } from '../utils'
+
+import { closeTerminalByEnvID } from './task'
+
+const LABEL_LIMIT = 15
+const MIME_TYPES_WITH_CUSTOM_RENDERERS = ['text/plain']
+
+export async function executeRunner(
+  runner: IRunner,
+  exec: NotebookCellExecution,
+  runningCell: TextDocument,
+  execKey: 'bash'|'sh',
+  environment?: IRunnerEnvironment
+) {
+  const cwd = path.dirname(runningCell.uri.fsPath)
+  
+  const RUNME_ID = `${runningCell.fileName}:${exec.cell.index}`
+  
+  const cellText = exec.cell.document.getText()
+  const script = getCmdShellSeq(cellText, PLATFORM_OS)
+
+  const annotations = getAnnotations(exec.cell)
+  const { interactive, mimeType, background } = annotations
+  
+  const program = await runner.createProgramSession({
+    // TODO: make this work on windows/if bash isn't installed/etc
+    programName: execKey,
+    environment,
+    script: {
+      type: 'script', script
+    },
+    envs: [
```

Comment:
I added this as part of the environment creation, so that this doesn't end up overwriting any changes to environment made by scripts. Let me know if you'd rather it be on every execution like you had it 👍